jira.tf

```diff
@@ -41,20 +68,48 @@ resource "aws_iam_role_policy_attachment" "jira" {
   policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
 }
 
+data "aws_secretsmanager_secret" "jira" {
+  count = var.enable_jira_integration ? 1 : 0
+  name  = var.jira_api_token_secret_name
+}
+
+data "aws_iam_policy_document" "secret_access" {
+  count = local.jira_api_token != null && var.create_jira_secret_access_policy ? 1 : 0
+
+  statement {
+    actions   = ["secretsmanager:GetSecretValue"]
+    effect    = "Allow"
+    resources = [local.jira_api_token]
+  }
+}
+
+resource "aws_iam_role_policy" "secret_access" {
+  count  = local.jira_api_token != null && var.create_jira_secret_access_policy ? 1 : 0
+  role   = aws_iam_role.jira[0].name
+  policy = data.aws_iam_policy_document.secret_access[0].json
+}
+
+
+
 resource "aws_lambda_function" "jira" {
   count            = var.enable_jira_integration ? 1 : 0
   function_name    = local.jira_name
-  filename         = data.archive_file.jira.output_path
-  handler          = "create_jira.handler"
+  filename         = "${path.module}/create_jira.zip"
```

Comment:
We don't have a rule for this but if I reference an interpolated string more than once I like it to be a local